lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1136,20 +1138,31 @@ static NodeValue simplifyConcatNode(Function *F, ConcatNode *CN) {
   return NodeValue(nullptr);
 }
 
-/// Optimize Concat nodes.
-static void optimizeConcatNodes(Function *F) {
-  auto &nodes = F->getNodes();
-
-  // For each node:
-  for (auto &node : nodes) {
-    if (auto *CN = dyn_cast<ConcatNode>(&node)) {
-      NodeValue newCN = simplifyConcatNode(F, CN);
-      if (newCN.getNode()) {
-        CN->getResult().replaceAllUsesOfWith(newCN);
+class OptimizeConcatNodes : public FunctionPass {
+public:
+  OptimizeConcatNodes(){};
+  ~OptimizeConcatNodes(){};
+  bool run(Function *F) override {
+    auto &nodes = F->getNodes();
+    bool changed = false;
+    // For each node:
+    for (auto &node : nodes) {
+      if (auto *CN = dyn_cast<ConcatNode>(&node)) {
+        NodeValue newCN = simplifyConcatNode(F, CN);
```

Comment:
would it make sense to move `simplifyConcatNode` to `private` method of OptimizeConcatNodes for better encapsulation?